daemon/routes/iprule/iprule.go

```diff
@@ -45,57 +46,66 @@ func (r *Router) SetupRoutingRules(
 		ipv6EnabledList = append(ipv6EnabledList, true)
 	}
 
-	defer func() { // recover if error
-		if err == nil {
-			return
-		}
-
-		for _, ipv6 := range ipv6EnabledList {
-			if err := removeSuppressprefixLengthRule(ipv6); err != nil {
-				log.Println(internal.DeferPrefix, err)
-			}
-
-			if err := removeFwmarkRule(r.fwmark, ipv6); err != nil {
-				log.Println(internal.DeferPrefix, err)
-			}
-		}
-	}()
-
 	for _, ipv6 := range ipv6EnabledList {
-		fwMarkRuleID, err := calculateRulePriority(ipv6)
-		if err != nil {
-			return err
-		}
-		routingTableID, err := calculateCustomTableID(ipv6)
+		fwmarkRulePresent, err := checkFwmarkRule(r.fwmark, ipv6)
 		if err != nil {
 			return err
 		}
-		r.tableID = routingTableID
-		if err = addFwmarkRule(
-			r.fwmark,
-			fwMarkRuleID,
-			routingTableID,
-			ipv6,
-		); err != nil {
-			return err
-		}
 
-		otherRuleID, err := calculateRulePriority(ipv6)
-		if err != nil {
-			return err
+		if !fwmarkRulePresent {
+			fwMarkRuleID, err := calculateRulePriority(ipv6)
+			if err != nil {
+				return err
+			}
+			routingTableID, err := calculateCustomTableID(ipv6)
+			if err != nil {
+				return err
+			}
+			r.tableID = routingTableID
+			if err = addFwmarkRule(
+				r.fwmark,
+				fwMarkRuleID,
+				routingTableID,
+				ipv6,
+			); err != nil {
+				return err
+			}
 		}
 
-		if err = addSuppressprefixLengthRule(
-			otherRuleID,
-			ipv6,
-		); err != nil {
-			return err
+		// PeerA (LAN-a 192.168.1.x) connects to PeerB (LAN-b 192.168.1.x)
+		// if PeerB allows its LAN access when used as Exit node
+		// then PeerB LAN access is the priority over PeerA LAN
+		if enableLocal {
+			enableLocalTraffic(ipv6)
```

Comment:
Some inconsistency - code above returns errors, this private function doesn't, and the private function below returns them but we are only logging it. I think instead of logging we should return all errors, especially because we are checking for existence of rules before acting, so any error would be unexpected.

Reply:
Agree